### ISSUE TITLE
Optionally ignore Kubernetes Ingress 'default backends' 

### ISSUE THREAD

**Issue by negz** [linkerd/linkerd#1793]
Issue Type:

- [ ] Bug report
- [x] Feature request

**Context**
Currently the `io.l5d.ingress` identifier determines the appropriate service for HTTP requests one of two ways:
* By matching explicit rules, derived from [Ingress resource rules](https://kubernetes.io/docs/concepts/services-networking/ingress/#the-ingress-resource), in its cache. These rules match on a host header, request path, or both.
* By matching '[default backends](https://kubernetes.io/docs/concepts/services-networking/ingress/#name-based-virtual-hosting)'. Default backends provide no parameters to route on, so by nature match all requests.

Given a cluster with multiple Ingress resources the ingress cache will iterate over all ingress resources, returning the first path that matches a rule. If no rules match, the ingress cache will iterate over all ingress resources again, returning the first one with a default backend.

**Proposed Change**
I'd like to avoid the [nondeterministic case](https://github.com/linkerd/linkerd/issues/1566#issuecomment-322246936) in which there are multiple default backends defined and a request matches no explicit rules. Instead all ingress requests should either:
1. Match an explicit rule.
1. Fall back to an explicitly configured service.

I believe this can be implemented by:
1. Adding a 'strict' mode in which the `io.l5d.ingress` identifier ignores default backends. i.e. If a request does not match any explicit rule the identifier fails.
1. Fall back to an `io.l5d.static` identifier that points to a Kubernetes service explicitly (or whatever you like really).

Together this would look something like:
```yaml
identifier:
  - kind: io.l5d.ingress
    strict: true
  - kind: io.l5d.static
    path: /coolnamespace/80/fallback
dtab: /svc => /#/io.l5d.k8s;
```

**Use Case**
At Planet Labs engineers have unfettered access to our clusters. They manage their own Ingress resources. We typically define one Ingress, one Service, and one Deployment resource for each Planet service. Historically we've used an in-house ingress controller that *only* supports default backends - no host or path rules. Requests with Host headers matching `ingressname.namespace.some.cluster.domain` are routed to said ingress's default backend.

Our newest generation of clusters use the linkerd ingress controller. We're concerned that engineers will deploy a default-backend-only Ingress resource, have things work at first, then have requests start to fail when another default backend is added.

**Comment by negz** [comment]
Other bugs where default backends have been gremlins:
* https://github.com/linkerd/linkerd/issues/1566
* https://github.com/linkerd/linkerd/issues/1187

**Comment by deebo91** [comment]
Thanks for the feature request @negz! The team will look at how we can add this feature to linkerd.

**Comment by negz** [comment]
To play devil's advocate on my own feature request: you could argue that this would be better enforced using an [admission control webhook](https://kubernetes.io/docs/admin/extensible-admission-controllers/#external-admission-webhooks) that prevents Ingress resources with default backends from being created than via a linkerd knob.

**Comment by deebo91** [comment]
Closing as a result of merged PR #1794
